otx/algorithms/classification/adapters/mmcls/data/pipelines.py

```diff
@@ -137,3 +139,64 @@ def __repr__(self):
         """Set repr of GaussianBlur."""
         repr_str = self.__class__.__name__
         return repr_str
+
+
+@PIPELINES.register_module()
+class PILImageToNDArray:
+    """Pipeline element that converts an PIL image into numpy array.
+
+    Expected entries in the 'results' dict that should be passed to this pipeline element are:
+        results['img']: PIL type image in data pipeline.
+
+    Args:
+        keys (list[str]): list to support multiple image converting from PIL to NDArray.
+    """
+
+    def __init__(self, keys=None):
+        self.keys = keys
+
+    def __call__(self, results):
+        """Callback function of PILImageToNDArray."""
+        for key in self.keys:
+            img = results[key]
+            img = np.asarray(img)
+            results[key] = img
+        return results
+
+    def __repr__(self):
+        """Repr function of PILImageToNDArray."""
+        repr_str = self.__class__.__name__
+        return repr_str
+
+
+@PIPELINES.register_module()
+class SeparateAug:
+    """Pipeline element that separates augmentation pipeline for mmcls.
+
+    Expected entries in the 'results' dict that should be passed to this pipeline element are:
+        results['img']: PIL type image in data pipeline.
+
+    Args:
+        pairs (dict): pairs to separate pipeline. key 'img' should be included.
+                      ex) dict(img=origin_pipeline, img_strong=strong_pipeline)
+    """
+
+    def __init__(self, pairs: dict):
+        if "img" not in pairs.keys():
+            raise ValueError("key 'img' should be included in pairs argumenent")
+        self.pipelines = {key: Compose(pipeline) for key, pipeline in pairs.items()}
+
+    def __call__(self, results):
+        """Callback function of SeparateAug."""
+        for key, pipeline in self.pipelines.items():
+            if key == "img":
+                output = pipeline(copy.deepcopy(results))
+            else:
+                output[key] = pipeline(results)["img"]
```

Comment:
From my point of view, the order of this is a bit confusing. Is it that `img` goes through the weak pipeline first, and then copying for strong img?

Reply:
No, `results` is **not augmented, original image**. 
output["img"] will be an image augmented by weak pipeline,
output[key] will be an image augmented by strong pipeline. This code does not result weak+strong (jointed in one) augmented image.

Reply:
So, isn't this method completely different from the last commit? Isn't it riding the common pipeline twice each?

Reply:
I understood that you wanted to have time independent data pipeline and merge into one element. Am I wrong?
What is your desired code? Please give me detailed description. I'll follow that.

Reply:
Reflected. Please review again for this part. I'll rebase after review.